lib/std/compress/zstandard.zig

```diff
@@ -216,43 +191,37 @@ pub fn DecompressStream(
                 }
             }
 
-            const size = @min(self.buffer.len(), buffer.len);
+            const size = @min(ring_buffer.len(), buffer.len);
             if (size > 0) {
-                self.buffer.readFirstAssumeLength(buffer, size);
+                ring_buffer.readFirstAssumeLength(buffer, size);
             }
-            if (self.state == .LastBlock and self.buffer.len() == 0) {
+            if (self.state == .LastBlock and ring_buffer.len() == 0) {
                 self.state = .NewFrame;
-                self.allocator.free(self.literal_fse_buffer);
-                self.allocator.free(self.match_fse_buffer);
-                self.allocator.free(self.offset_fse_buffer);
-                self.allocator.free(self.literals_buffer);
-                self.allocator.free(self.sequence_buffer);
-                self.buffer.deinit(self.allocator);
             }
             return size;
         }
     };
 }
 
 pub fn decompressStreamOptions(
-    allocator: Allocator,
     reader: anytype,
     comptime options: DecompressStreamOptions,
-) DecompressStream(@TypeOf(reader, options)) {
-    return DecompressStream(@TypeOf(reader), options).init(allocator, reader);
+    window_buffer: *[options.window_size_max]u8,
+) DecompressStream(@TypeOf(reader), options) {
+    return DecompressStream(@TypeOf(reader), options).init(reader, window_buffer);
 }
```

Comment:
This makes it possible for the user to pass a buffer that does not have size equal to `options.window_size_max`, which will lead to a compile error. Perhaps it would be better to make a separate options struct that does not include the window size. Doing this would require the new type to be kept in sync with any future changes to `DecompressStreamOptions`.